package Configuration;

message units{

required float aperture = 1;
required float pca_scl = 2;
repeated float pca = 3 [packed=true];
repeated float templates = 4 [packed=true];

}

message channels{
  
  required int32 id = 1;
  required float gain = 2;
  required float agc = 3;
  required float centering = 4;
  required float threshold = 5;
  repeated units unit = 6;
  
  repeated float pca_mean = 7 [packed=true];
  repeated float pca_max = 8 [packed=true];
  
 }
  
message parameters{

  required int32 signal_chain = 1;
  repeated channels channel = 2;
  repeated int32 selected = 3;

}

message state{

	optional string default_configuration = 1;
}
